x-pack/plugin/eql/src/main/antlr/EqlBase.g4

```diff
@@ -98,18 +103,19 @@ primaryExpression
     : constant                                                                          #constantDefault
     | functionExpression                                                                #function
     | qualifiedName                                                                     #dereference
+    | ESCAPED_IDENTIFIER                                                                #identifierEscape
```

Comment:
Why not declare both `ESCAPED_IDENTIFIER` and `IDENTIFIER` under `identifier` as in the previous version and use `IDENTIFIER` where needed (such as function names - it's might be worth using STRING or declaring a different alphanumeric for that)?

Reply:
Yeah, function names and pipe names should be the only exceptions to that.
Would you prefer to support both of these? I figured that if you are going to escape with `, then you might as well escape the whole thing, but I'm not attached:
```sql
# ticks around the full path. dots interpreted as normal
`some-escaped-field.one-more.escaped-!-field`

# ticks around each key
`some-escaped-field`.`one-more`.`escaped-!-field`

Reply:
Should the `functionExpression` rule pick a pipe/function declaration? In general I'd rather have us use qualifiedName everywhere (instead of identifier) so that user are able to use the `.` hierarchy and thus avoid redundancy (qualifiedName & identifier).